include/swift/AST/DiagnosticsSema.def

```diff
@@ -1054,9 +1054,10 @@ ERROR(missing_dynamic_callable_kwargs_method,none,
       "@dynamicCallable type %0 cannot be applied with keyword arguments; "
       "missing 'dynamicCall(withKeywordArguments:)' method", (Type))
 
-ERROR(type_invalid_dml,none,
+ERROR(invalid_dynamic_member_lookup_type,none,
       "@dynamicMemberLookup attribute requires %0 to have a "
-      "'subscript(dynamicMember:)' member with a string index", (Type))
+      "'subscript(dynamicMember:)' method with an "
+      "'ExpressibleByStringLiteral' argument", (Type))
```

Comment:
There are a few names for "argument to subscript method": "argument"/"parameter" and "index".
I think "argument" is a bit easier to understand (unmistakable meaning).

Reply:
Here, I _think_ the right term would be “parameter,” would it not? As in, one passes arguments to methods that have parameters.

Reply:
Yes, I agree. 😃
Renamed "argument" to "parameter" in https://github.com/apple/swift/pull/20498/commits/18a8ea911e647867c392829bdda94ca62a48855e.